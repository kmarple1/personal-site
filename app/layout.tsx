import type { Metadata } from "next";
import { Inter } from "next/font/google";
import Script from "next/script";

import { ThemeProvider } from "next-themes";

import InnerLayout from "./innerLayout";

import "./globals.css";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Kyle Marple - Personal Site and Portfolio",
  description: "A personal site to serve as a resume and portfolio for Kyle Marple (me)",
};

// istanbul ignore next: RTL won't support testing this component because of the <html> wrapper
const RootLayout = ({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) => {
  return (
    <html lang="en" suppressHydrationWarning>
      <body className={`antialiased ${inter.className}`}>
        {/* next-themes uses the color-scheme CSS property to differentiate light and dark themes,
        but Tailwind is watching for a CSS class. */}
        <Script id="theme-detector">{`
          const theme = document.documentElement.style.colorScheme
          document.documentElement.classList.add(theme)
        `}</Script>
        <ThemeProvider>
          <InnerLayout>{children}</InnerLayout>
        </ThemeProvider>
      </body>
    </html>
  );
};

export default RootLayout;
