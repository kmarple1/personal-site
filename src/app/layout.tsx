import type { Metadata } from "next";
import { Inter } from "next/font/google";
import Script from "next/script";
import { ThemeProvider } from "next-themes";
import Header from "./components/header/header";
import "./globals.css";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Kyle Marple - Personal Site and Portfolio",
  description: "A personal site to serve as a resume and portfolio for Kyle Marple (me)",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en" suppressHydrationWarning>
      <body className={`antialiased ${inter.className}`}>
        {/* next-themes uses the color-scheme CSS property to differentiate light and dark themes,
        but Tailwind is watching for a CSS class. */}
        <Script id="theme-detector">{`
          const theme = document.documentElement.style.colorScheme
          document.documentElement.classList.add(theme)
        `}</Script>
        <ThemeProvider>
          <div className="flex h-full w-full flex-col bg-base-100 text-base-content">
            <Header />
            <div className="w-full">
              <main className="mx-auto max-w-8xl p-4 sm:px-6 md:px-8">{children}</main>
            </div>
          </div>
        </ThemeProvider>
      </body>
    </html>
  );
}
