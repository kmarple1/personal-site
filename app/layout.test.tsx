import { render, screen, waitFor } from "@testing-library/react";

import { Layout, metadata } from "./layout";
import Home from "./page";

describe("Root Layout", () => {
  it("renders without error", async () => {
    render(
      <Layout>
        <Home />
      </Layout>,
    );

    await waitFor(() => {
      expect(screen.getByText("Theme")).toBeInTheDocument();
    });

    expect(screen.getByText("Kyle Marple - Personal Site")).toBeInTheDocument();
  });

  it("has expected metadata", () => {
    expect(metadata).toMatchSnapshot();
  });
});
