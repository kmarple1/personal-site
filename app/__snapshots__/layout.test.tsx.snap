// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Layout has expected metadata 1`] = `
{
  "description": "A personal site to serve as a resume and portfolio for Kyle Marple (me)",
  "title": "Kyle Marple - Personal Site and Portfolio",
}
`;
